openstack/db/v1/instances/results.go

```diff
@@ -56,6 +57,33 @@ type Instance struct {
 	Datastore datastores.DatastorePartial
 }
 
+func (s *Instance) UnmarshalJSON(b []byte) error {
```

Comment:
I think the logic here is sound, but let's make it conform to the other `UnmarshalJSON` methods. See [here](https://github.com/gophercloud/gophercloud/blob/master/openstack/orchestration/v1/stacks/results.go#L62) for an example.